<launch>
  <arg name="cam_name" value="my_camera" />

  <!-- Start the GSCAM node -->
  <env name="GSCAM_CONFIG" 
      value="v4l2src device=/dev/video1 ! video/x-raw-rgb,framerate=30/1,width=640,height=480 ! ffmpegcolorspace " />
  <node pkg="gscam" type="gscam" name="$(arg cam_name)">
    <param name="camera_name" value="$(arg cam_name)" />
    <remap from="camera/image_raw" to="$(arg cam_name)/image_raw" />
  </node>

  <!-- Start our service node and opencv node -->
  <node pkg = "assignment_5" type = "motion_mode_keyboard" name = "motion_mode_keyboard"
    output="screen" launch-prefix="gnome-terminal -e" />
  <node pkg = "assignment_5" type = "motion_detector" name = "motion_detector"
    output="screen" />


    <node pkg="image_view" type="image_view" name="raw_image" >
        <remap from="image" to="/$(arg cam_name)/image_raw" />
    </node>
    
    <node pkg="image_view" type="image_view" name="image_motion_boxed" >
        <remap from="image" to="/$(arg cam_name)/image_motion_boxed" />
    </node>

</launch>
